frontend/src/components/Pages/InspectionPage/ScheduleMissionDialog.tsx

```diff
@@ -96,40 +149,102 @@ export const ScheduleMissionDialog = (props: IProps): JSX.Element => {
-                            </Button>
-                            <Button
-                                onClick={() => {
-                                    onScheduleButtonPress()
-                                    props.closeDialog()
-                                }}
-                                disabled={!selectedRobot}
-                            >
-                                {' '}
-                                {TranslateText('Queue mission')}
-                            </Button>
-                        </StyledMissionSection>
-                    </StyledAutoComplete>
+                    <StyledDialogContent>
+                        <Typography variant="h4">{TranslateText('Add mission')}</Typography>
+                        {isAlreadyScheduled && (
+                            <StyledDangerContent>
+                                <Icon name={Icons.Warning} size={16} color="red" />
+                                <Typography variant="body_short" color="red">
+                                    {TranslateText('The mission is already in the queue')}
+                                </Typography>
+                            </StyledDangerContent>
+                        )}
+                        <StyledAutoComplete>
+                            <Autocomplete
+                                optionLabel={(r) => r.name + ' (' + r.model.type + ')'}
+                                options={robotOptions}
+                                label={TranslateText('Select robot')}
+                                onOptionsChange={(changes) => onSelectedRobot(changes.selectedItems[0])}
+                                autoWidth={true}
+                                onFocus={(e) => e.preventDefault()}
+                            />
+
+                            <StyledMissionSection>
+                                <Button
+                                    onClick={() => {
+                                        props.closeDialog()
+                                    }}
+                                    variant="outlined"
+                                    color="primary"
+                                >
+                                    {' '}
+                                    {TranslateText('Cancel')}{' '}
+                                </Button>
+                                <Button
+                                    onClick={() => {
+                                        onScheduleButtonPress()
+                                        props.closeDialog()
+                                    }}
+                                    disabled={!selectedRobot}
+                                >
+                                    {' '}
+                                    {TranslateText('Queue mission')}
+                                </Button>
+                                {isAlreadyScheduled && props.missions.length > 1 && (
```

Comment:
```suggestion
                                {isAlreadyScheduled && unscheduledMissions > 1 && (
```
Should it be this? Right now the button is there even when all the missions on the deck is scheduled

Reply:
unscheduledMissions.length I mean

Reply:
This is just because I am using the same dialog when scheduling in the table and the button on the deck, and for the table I want to hide the only schedule missions that are not in the queue button